docs/path-resolution.rst

```diff
@@ -373,6 +373,72 @@ The resulting file path becomes the source unit name.
     When working with older versions of the compiler it is recommended to invoke the compiler from
     the base path and to only use relative paths on the command line.
 
+.. index:: ! allowed paths, ! --allow-paths, remapping; target
+.. _allowed-paths:
+
+Allowed Paths
+=============
+
+As a security measure, the Host Filesystem Loader will refuse to load files from outside of a few
+locations that are considered safe by default:
+
+- Outside of Standard JSON mode:
+
+  - The directories containing input files listed on the command line.
+  - The directories used as :ref:`remapping <import-remapping>` targets.
```

Comment:
Wait what do remapping targets have to do with the host filesystem?

Reply:
They end up in source unit names so ultimately they can get passed to the callback. For example in

```bash
echo "import 'dir/contract.sol';" | solc test.sol dir/=anotherdir/ -
```

you'll get an error about `anotherdir/contract.sol` not being allowed if we do not whitelist the targets.

This is how it already worked so I preserved it but I do find it a bit weird not to require people to whitelist them explicitly. Especially when a remapping is not a full directory name (e.g. `dir=anoth`) or when it's not meant to be path at all. I'd be for removing that but I think it's breaking.

Reply:
I think it might have something to do with how remappings on the command line without the `=<target>` part are supposed to be equivalent to input files.

Reply:
OK, so it was like that before? Then it is fin, but we can consider changing it with 0.9.0.

Reply:
Yes, here's what the code was doing in 0.8.0 long before I started changing the CLI and path-related stuff:
https://github.com/ethereum/solidity/blob/c7dfd78e57c5ad7abd485dc1cc13d8f0ab09d431/solc/CommandLineInterface.cpp#L585-L589
https://github.com/ethereum/solidity/blob/c7dfd78e57c5ad7abd485dc1cc13d8f0ab09d431/solc/CommandLineInterface.cpp#L631

Reply:
#12014